### ISSUE TITLE
Does simpletransformers train word embeddings on the fly?

### ISSUE THREAD

**Issue by noobistz** [ThilinaRajapakse/simpletransformers#223]
Hi, I have been experimenting with the simpletransformers library and could not find it in the documentation. Does it train word embeddings on the fly or uses pre-trained embeddings? If it's the latter, isit possible to train my own word embeddings and use that instead? How can I go about doing so?

**Comment by ThilinaRajapakse** [comment]
It is _sort of_ done on the fly. But it is not a direct mapping from words to embeddings as in classic word embeddings. Training your own word embeddings would actually mean training your own Transformer model which is generally impractical. When you fine-tune a pre-trained model, you are basically teaching the model to create better representations for samples your particular dataset.

**Comment by noobistz** [comment]
Thank you for your clarification.

I have successfully trained a NERmodel on my own dataset. However, when i used the eval_model() function on my test data of 20989 words with ground truth labels, i am only getting a total of 13319 predicted labels. I have looked into #11, but it didn't seem to resolve the problem.

**Comment by ThilinaRajapakse** [comment]
This is possibly due to #234 . Can you try it now and see if it is fixed?

**Comment by noobistz** [comment]
I updated the package via pip to 0.20.1, loaded the saved model and tried eval_model() but the output is still the same.

**Comment by ThilinaRajapakse** [comment]
Can you copy just the evaluation script, the data file, and the outputs directory to a new directory and run it from there? Just to rule out cache issues entirely.

**Comment by noobistz** [comment]
I have just tried doing it and the outcome remains the same.

**Comment by ThilinaRajapakse** [comment]
Can you give me a minimal script (with minimal data) to reproduce the issue?

**Comment by noobistz** [comment]
I have created a repository with sample code and data to reproduce the issue (link below). The output is slightly different but the problem of different number of predicted labels vs total test data length still exists. (46417 vs 46665).

[Code](https://github.com/noobistz/simpletransformers)
